src/image.rs

```diff
@@ -95,9 +89,8 @@ pub fn get_width(mut cx: FunctionContext) -> JsResult<JsValue> {
 pub fn set_width(mut cx: FunctionContext) -> JsResult<JsUndefined> {
   let this = cx.argument::<BoxedImage>(0)?;
   let mut this = this.borrow_mut();
-  if let Some(num) = opt_float_arg(&mut cx, 1){
-    this.size.width = i32::max(0, num as i32);
-  }
+  let num = float_arg_or(&mut cx, 1, 0.0);
```

Comment:
Why default to zero if it's NaN?

Reply:
Actually now that I think about it, if it's null/undefined then the size could perhaps be reset to default. Although browsers just ignore anything non-numeric here. The way I had it before the user could reset it with a negative value, but that's not intuitive.

Reply:
When I tested on Safari, assigning an actual `NaN` or any non-Number type actually gets interpreted as `0` rather than just ignored. Even assigning `undefined` will be a zero (and not a reset)

Reply:
[not that this would be the first time Safari did things differently than every other browser…]

Reply:
> this seems to be what browsers do when you assign non-numbers to .width and .height

Just saw this and ran some more tests on FF and Chrome.  And you're right!  Though they also `parseFloat()` the value first so a numeric string does still work. Still kinda like my reset idea, but I suppose it's not necessary.

Reply:
Ah we're cross posting.  Safari is the new IE so indeed I wouldn't base much on that. But yes Chrome and FF act the same (except perhaps with numeric strings).

Reply:
Oh shoot, you're right. Strings that are parsable *should* work. Lemme fix that…

Reply:
> reset idea

It's not as pithy but I guess the W3C's idea for the 'reset' is `img.width = img.naturalWidth`?

Reply:
> img.width = img.naturalWidth

Well that sets it to the current natural width... if one, for some reason, sets a new `src` on the Image, it may no longer be valid. But yea probably no one needs that IRL.